crates/storage/provider/src/providers/database/provider.rs

```diff
@@ -1140,6 +1141,65 @@ impl<TX: DbTx> BlockReader for DatabaseProvider<TX> {
     }
 }
 
+impl<TX: DbTx> TransactionsProviderExt for DatabaseProvider<TX> {
+    /// Recovers transaction hashes by walking through `Transactions` table and
+    /// calculating them in a parallel manner. Returned unsorted.
+    fn transaction_hashes_by_range(
+        &self,
+        tx_range: Range<TxNumber>,
+    ) -> RethResult<Vec<(TxHash, TxNumber)>> {
+        let mut tx_cursor = self.tx.cursor_read::<tables::Transactions>()?;
+        let tx_range_size = tx_range.clone().count();
+        let tx_walker = tx_cursor.walk_range(tx_range)?;
+
+        let chunk_size = (tx_range_size / rayon::current_num_threads()).max(1);
```

Comment:
I think this can be problematic, because this can negatively impact everything else.

we're usually calling this on a per-block basis, right?

so in worst case we're spawning num CPU parallel worker threads with >1 txs.

the cutoff should be way higher so that the overhead of using the threadpool is worth it.

do we have benchmarks for this? 

let's say we have 200 tx in a block and 16cores results in a chunk size of 12, which I think is a bit too low.

I think we should rather define a minimum txs per thread, maybe go with 32 or something until we have benchmarks, then we also need a variant where we don't even need to spawn a single thread.

but we can do all of this in a followup if we track it

Reply:
https://github.com/paradigmxyz/reth/issues/5190